README.md

```diff
@@ -1,15 +1,33 @@
 # cuda-fixnum for snark challenge
 
-For each of mnt4 and mnt6, this takes the pairwise product of two arrays. That is, it maps over two arrays.
+Use this code to get started on the snark challenge. It implements the logic needed to do field arithmetic. In particular, for the fields used by mnt4-753 and mnt6-753, this takes the pairwise product of two arrays of field elements. That is, it maps over two arrays.
 
 See `main.cu` for the implementation
 
+## Suggested steps to make a submission for the snark challenge
+
+Here are some suggested steps to solve the tutorial stage and work towards a faster, GPU powered snark prover:
+
+1. [Add Quadratic extension arithmetic (Tutorial, $150)](https://coinlist.co/build/coda/pages/problem-02-quadratic-extension-arithmetic)
+2. [Add Cubic extension arithmetic (Tutorial, $150)](https://coinlist.co/build/coda/pages/problem-03-cubic-extension-arithmetic)
+3. [Add Curve operations (Tutorial, $200)](https://coinlist.co/build/coda/pages/problem-04-curve-operations)
+
+For each of these, the cuda kernel code will need to be changed (see `main.cu:21-35`)
+
+And here is our best guess on how to effectively make a submission for the [full prover](https://coinlist.co/build/coda/pages/problem-07-groth16-prover-challenges) (up to $70,000, and $7,000 immediately for the first submission to 2x the speed)
+
+The snark prover is composed of several FFTs and multiexponentiations.
+
+1. Once you've finished the tutorial, try swapping out the multi-exponentiations in the prover for on-gpu versions using the curve operations from the tutorial. This may already lead to a big speed improvement. 
+2. Do the multi-exponentiation component entirely on-gpu, using an on-gpu [reduce](https://github.com/NVIDIA/cuda-samples/tree/master/Samples/reduction)
```

Comment:
It's not exactly clear to me how this comment relates to the previous one. Doesn't the previous comment already suggest replacing the whole multi-exp? Maybe the previous thing should somehow explain how to do a map exp and then do the multi part on cpu?

Reply:
gp. how would you phrase the different components of the the multi-exponentiation? Like saying, replace each exponentiation with on-gpu versions (map), and then combine the results for each exponentiation on-gpu? (reduce)

Reply:
Maybe something like, 

Each multi-exponentiation can be seen as a [map-reduce, as explained here](https://youtu.be/81uR9W5PZ5M). The reduce part may be complicated to implement for GPU, so it may be a good idea to start by implementing the "map" part on GPU by using the [starter "map" code here](https://github.com/es92/cuda-fixnum/) and the "reduce" part on CPU.

or some such.
The CPU implementation however doesn't have a neat division like this.